bot/exts/moderation/slowmode.py

```diff
@@ -66,37 +89,98 @@ async def set_slowmode(
         humanized_delay = time.humanize_delta(delay)
 
         # Ensure the delay is within discord's limits
-        if slowmode_delay <= SLOWMODE_MAX_DELAY:
-            log.info(f"{ctx.author} set the slowmode delay for #{channel} to {humanized_delay}.")
-
-            await channel.edit(slowmode_delay=slowmode_delay)
-            if channel.id in COMMONLY_SLOWMODED_CHANNELS:
-                log.info(f"Recording slowmode change in stats for {channel.name}.")
-                self.bot.stats.gauge(f"slowmode.{COMMONLY_SLOWMODED_CHANNELS[channel.id]}", slowmode_delay)
+        if not slowmode_delay <= SLOWMODE_MAX_DELAY:
+            log.info(
+                f"{ctx.author} tried to set the slowmode delay of #{channel} to {humanized_delay}, "
+                "which is not between 0 and 6 hours."
+            )
 
             await ctx.send(
-                f"{Emojis.check_mark} The slowmode delay for {channel.mention} is now {humanized_delay}."
+                f"{Emojis.cross_mark} The slowmode delay must be between 0 and 6 hours."
             )
+            return
 
-        else:
+        if duration is not None:
+            slowmode_duration = time.relativedelta_to_timedelta(duration).total_seconds()
+            humanized_duration = time.humanize_delta(duration)
+
+            expiration_time = datetime.now(tz=UTC) + timedelta(seconds=slowmode_duration)
+            expiration_timestamp = discord_timestamp(expiration_time, TimestampFormats.RELATIVE)
+
+            # Only update original_slowmode_cache if the last slowmode was not temporary.
+            if not await self.slowmode_expiration_cache.contains(channel.id):
+                await self.original_slowmode_cache.set(channel.id, channel.slowmode_delay)
+            await self.slowmode_expiration_cache.set(channel.id, expiration_time.timestamp())
+
+            self.scheduler.schedule_at(expiration_time, channel.id, self._revert_slowmode(channel.id))
             log.info(
-                f"{ctx.author} tried to set the slowmode delay of #{channel} to {humanized_delay}, "
-                "which is not between 0 and 6 hours."
+                f"{ctx.author} set the slowmode delay for #{channel} to"
+                f"{humanized_delay} which expires in {humanized_duration}."
```

Comment:
```suggestion
                f"{humanized_delay} which expires {humanized_duration}."
```
Same here, can specify what it's going to revert to.